packages/contracts/src/dollar/libraries/LibBancorFormula.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import {BancorPower} from "../core/BancorPower.sol";
+import "abdk/ABDKMathQuad.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "./Constants.sol";
+
+/**
+ * @title Bancor formula
+ * @dev Modified from the original
+ * @notice forked from Bancor:
+ * https://github.com/bancorprotocol/contracts
+ *
+ */
+library LibBancorFormula {
+    using ABDKMathQuad for uint256;
+    using ABDKMathQuad for bytes16;
+
+    /**
+     * @dev Given a token supply, reserve balance, weight and a deposit amount (in the reserve token),
+     * calculates the target amount for a given conversion (in the main token)
+     *
+     * @dev _supply * ((1 + _tokensDeposited / _connectorBalance) ^ (_connectorWeight / 1000000) - 1)
+     *
+     * @param _tokensDeposited   amount of collateral tokens to deposit
+     * @param _connectorWeight   connector weight, represented in ppm, 1 - 1,000,000
+     * @param _supply          current Token supply
+     * @param _connectorBalance   total connector balance
+     * 
+     * @return amount of Tokens minted
+     */
+    function _purchaseTargetAmount(
```

Comment:
we can merge this lib with the bonding curve lib

Reply:
Done